### PR TITLE
[RW-6930][risk=low] Populate AccessModules in Profile API

### PR DESCRIPTION
**Pull request by yonghaoy**
Description:
This will replace 
All AAR stuff, including the yaml's model, and some logic in UserServiceImpl, e.g.: 
https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java#L240-L314

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java:35]: since we have now passed this time, it's probably better to delete it.

File: api/src/main/java/org/pmiops/workbench/access/UserAccessModuleMapper.java
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/UserAccessModuleMapper.java:23]: why is this one handled differently?

Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/UserAccessModuleMapper.java:23]: please add a note indicating that the behavior here is different because it's a derived value

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java:35]: I keep it because I am not sure if that will break things. Code I am not sure is: https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java#L270
If removing  MIN_ACCESS_EXPIRATION_EPOCH_MS, API may return different value(e.g. anytime before that would becomes the really expiration time instead of the MIN_ACCESS_EXPIRATION_EPOCH_MS). 
If our UI does not use that value to render information, or we are fine to use the older timestamp, then we can remove that.

Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java]: I think it's probably OK to remove... but I'll do it in a standalone PR so we can analyze it in isolation

File: api/src/main/resources/workbench-api.yaml
Comment (jmthibault79) [inline api/src/main/resources/workbench-api.yaml:5688]: this text is no longer true.  Are you planning to keep the idea of `hasExpired` or remove it?

File: api/src/main/resources/workbench-api.yaml
Comment (jmthibault79) [inline api/src/main/resources/workbench-api.yaml:5716]: add "if applicable" or similar to all of these

File: api/src/test/java/org/pmiops/workbench/profile/ProfileServiceTest.java
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/profile/ProfileServiceTest.java:113]: unused - remove

File: api/src/main/java/org/pmiops/workbench/access/AccessUtils.java
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessUtils.java:13]: We should really merge these two enums.  Is the only thing preventing this the difference between `DATA_USE_AGREEMENT` and `DATA_USER_CODE_OF_CONDUCT`?  If so, PR #5251 fixes that.

**Review by jmthibault79 — CHANGES_REQUESTED** [review]
tests are failing.  Please rebase and fix tests.

File: api/src/main/resources/workbench-api.yaml
Comment (yonghaoy) [reply api/src/main/resources/workbench-api.yaml:5688]: updated

File: api/src/main/java/org/pmiops/workbench/access/AccessUtils.java
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/access/AccessUtils.java:13]: Also COMPLIANCE_TRAINING -> RT_COMPLIANCE_TRAINING
RAS_LINK_LOGIN_GOV -> RAS_LOGIN_GOV.
Might do that in separate since that field also used in UI, and after 5251 merged...

**Comment by yonghaoy** [comment]
> tests are failing. Please rebase and fix tests.

Test are passing now. PTAL, thanks

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java:145]: I don't know if extract is the right word here.  Derive maybe?

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java:130]: I like this separation, but I think we actually want `||` for all of these.  I'd also prefer to have the `enableAccessRenewal` check first.

File: api/src/main/java/org/pmiops/workbench/access/AccessUtils.java
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/access/AccessUtils.java:16]: Yes it can certainly wait for a different PR

File: api/src/main/java/org/pmiops/workbench/profile/ProfileService.java
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/profile/ProfileService.java:154]: use `clock.instant()` instead of `Instant.now()` - this makes test injection much easier

File: api/src/main/resources/workbench-api.yaml
Comment (jmthibault79) [reply api/src/main/resources/workbench-api.yaml]: Thanks!  Could you update it a little further, something like: "Whether any modules have expired. Expiration information is also included with each module in the array; this field is provided for convenience."

File: api/src/test/java/org/pmiops/workbench/access/AccessModuleServiceTest.java
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/access/AccessModuleServiceTest.java:236]: yeah.  but it's ok.

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java:130]: Please add an AccessModuleServiceTest for the case where enableAccessRenewal=false

File: api/src/test/java/org/pmiops/workbench/access/AccessModuleServiceTest.java
Comment (yonghaoy) [reply api/src/test/java/org/pmiops/workbench/access/AccessModuleServiceTest.java:217]: swapped with RT_training

**Comment by calbach** [comment]
I'll let @jmthibault79 take this one, @yonghaoy let me know if you wanted me to look at anything specific

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java]: Good catch! Also added a test for that.

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java:145]: spelling: derive

**Review by jmthibault79 — APPROVED** [review]
Good to go after:
* tests pass
* "derive" spelling change
* Instant.now() -> clock.instant()

**Pull request merged**
